src/main/java/pageObjects/Licenses.java

```diff
@@ -0,0 +1,264 @@
+package pageObjects;
+
+import driver.DriverFactory;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import utils.Constants;
+import utils.RandomGenerator;
+
+import static utils.Constants.*;
+
+public class Licenses extends DriverFactory {
+    public Licenses(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    @FindBy(id = "undefined-licenses-mButton")
+    private WebElement licenseButton;
+    @FindBy(id = "add-license")
+    private WebElement addLicenseButton;
+
+    @FindBy(id = "license_name")
+    private WebElement nameField;
+    @FindBy(id = "item_save")
+    private WebElement saveLicense;
+    @FindBy(id = "license_purchase_date_picker")
+    private WebElement purchaseDate;
+    @FindBy(id = "license_expiration_date_picker")
+    private WebElement expirationDate;
+    @FindBy(id = "license_type-selectList")
+    private WebElement licenseTypeList;
+    @FindBy(id = "license_type-selectList-pack")
+    private WebElement selectPackList;
+    @FindBy(id = "license_type-selectList-individual")
+    private WebElement selectIndividualList;
+    @FindBy(id = "license-locations-selectList")
+    private WebElement licenseLocationList;
+    @FindBy(id= "license-locations-selectList-romania")
+    private WebElement RomaniaList;
+    @FindBy(id = "license_dropdown_11111")
+    private WebElement licenceDropdown;
+    @FindBy(id = "license_edit_11111")
+    private WebElement licenceEditDropdown;
+
+    public void checkIfUserCanAddLicense() {
+        signInAsUser();
+        licenseButton.click();
+    }
+    public void successfulCreateLicense() {
+        signInAsAdmin();
+        licenseButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("add-license")));
+        addLicenseButton.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSE_DETAILS_URL));
+        nameField.sendKeys(LICENSE_NAME_WS);
+        purchaseDate.sendKeys((PURCHASE_DATE_WS));
+        expirationDate.sendKeys((EXPIRATION_DATE_WS));
+        saveLicense.click();
+        wait.until(ExpectedConditions.urlToBe(Constants.LICENSES_URL));
+    }
+
+    public void checkIfNameFieldIsMandatory() {
+        signInAsAdmin();
+        licenseButton.click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("add-license")));
+        addLicenseButton.click();
```

Comment:
if we already use a webelement called addLicenseButton with the id add-license, wouldnt it be better if we just call on the visibility of that element instead of finding it by id every time? (replace everywhere you see fit in your PR)